language: python

sudo: false

python:
    - "2.7"
    - "3.2"
    - "3.3"
    - "3.4"
    - "3.5"
    - "3.6"

env:
    - DJANGO="Django>=1.8,<1.9"
    - DJANGO="Django>=1.9,<1.10"
    - DJANGO="Django>=1.10,<1.11"
    - DJANGO="Django>=1.11,<2.0"

matrix:
    exclude:
        - python: "3.6"
          env: DJANGO="Django>=1.8,<1.9"
        - python: "3.2"
          env: DJANGO="Django>=1.9,<1.10"
        - python: "3.3"
          env: DJANGO="Django>=1.9,<1.10"
        - python: "3.6"
          env: DJANGO="Django>=1.9,<1.10"
        - python: "3.2"
          env: DJANGO="Django>=1.10,<1.11"
        - python: "3.3"
          env: DJANGO="Django>=1.10,<1.11"
        - python: "3.6"
          env: DJANGO="Django>=1.10,<1.11"
        - python: "3.2"
          env: DJANGO="Django>=1.11,<2.0"
        - python: "3.3"
          env: DJANGO="Django>=1.11,<2.0"

addons:
    apt:
        packages:
            - libgeoip-dev

install:
    - pip install "$DJANGO"
    - pip install geoip
    - pip install coverage django-discover-runner mock unittest2
    - pip freeze
    - python setup.py develop

script:
    - python example/manage.py test --verbosity=2
